src/daemon/mod.rs

```diff
@@ -393,21 +391,32 @@ pub(super) async fn start(
         &config.client.data_dir,
     );
 
+    // TODO: respect `--consume-snapshot` CLI option once it's implemented
+    let mut consume_snapshot_file = false;
+    // Sets the latest snapshot if needed for downloading later
     let mut config = config;
-    fetch_snapshot_if_required(&mut config, epoch, opts.auto_download_snapshot).await?;
-
-    if let Some(path) = &config.client.snapshot_path {
-        let stopwatch = time::Instant::now();
-        import_chain(
-            &state_manager,
-            &path.display().to_string(),
-            config.client.skip_load,
-            config.client.chunk_size,
-            config.client.buffer_size,
+    if config.client.snapshot_path.is_none() {
+        set_snapshot_path_if_needed(
+            &mut config,
+            epoch,
+            opts.auto_download_snapshot,
+            &db_root_dir,
         )
-        .await
-        .context("Failed miserably while importing chain from snapshot")?;
-        info!("Imported snapshot in: {}s", stopwatch.elapsed().as_secs());
+        .await?;
+        consume_snapshot_file = true;
```

Comment:
No need to set this flag as true. All downloaded files are always consumed.

Reply:
Removed